### ISSUE TITLE
dnsperf doesn't reconnect in TCP mode

### ISSUE THREAD

**Issue by UladzimirTrehubenka** [Sinodun/dnsperf-tcp#6]
I am trying to use dnsperf (build on Fedora 24) against [CoreDNS](https://github.com/coredns/coredns).
CoreDNS uses Miek Gieben [dns library](https://github.com/miekg/dns).
The library has limitation in 128 TCP packets (see [listing](https://github.com/miekg/dns/blob/master/server.go#L541)), after reach that TCP session is closed.
So command
`dnsperf -s 127.0.0.1 -p 53 -d dns.txt -Q 10000 -l 10 -z`
failed with message
`Warning: Error: cannot use connection: Connection reset by peer`

**Comment by saradickinson** [comment]
Yeah, this error isn't currently handled correctly - there is a TODO in the code to fix this, so thanks for logging the issue.
As a workaround you can use the command line flag -Z to set the number of queries dnsperf sends on each connection so as long as this number is less than 128 then you shouldn't hit this issue.
